### PR TITLE
feat(search-with-typeahead): v2 styles for search-with-typeahead

### LINKED ISSUE #10790: [Component] Search with type ahead
### Design changes
**Alternate**
- [ ] Remove customized styling in the `alternate` style
- [ ] Update the `alternate` search with typeahead to use the carbon web components default search in `large` size directly

Image

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

Closes #10790

### Description

Fixes the styles for `search-with-typeahead` and brings the `alternate` variation down to 48px in height

### Changelog

**Changed**

- add `sticky-search` property to dotcom-shell story variation with the `leadspace-with-search` to be able to test out the scrolling

### CODE REVIEW

**Review by IgnacioBecerra — APPROVED** [review]
LGTM!

**Review by emyarod — APPROVED** [review]
looks good to me!

File: packages/styles/scss/components/search-with-typeahead/_search-with-typeahead.scss
Comment (kennylam) [inline packages/styles/scss/components/search-with-typeahead/_search-with-typeahead.scss:1]: @annawen1 do we need any of these React styles anymore?

Comment (annawen1) [reply packages/styles/scss/components/search-with-typeahead/_search-with-typeahead.scss:1]: yeah despite the `react` in the class names, it's being used in the web component classes

Screenshot 2023-09-01 at 11 40 03 AM

Comment (annawen1) [reply packages/styles/scss/components/search-with-typeahead/_search-with-typeahead.scss:1]: I can do a clean up job later on to remove the `react` from the class names

Comment (kennylam) [reply packages/styles/scss/components/search-with-typeahead/_search-with-typeahead.scss:1]: Yeah, sounds good!

**Comment by oliviaflory** [comment]
@annawen1 overall this looks really good! I saw a few things:

### Default
#### - [ ] Blue line isn't accessible in dark themes
Let me know if  you need help with a token for this
Screenshot 2023-09-01 at 3 31 42 PM

#### - [ ]  Text indicator (vertical cursor) and text when typing into search box should be `$text-primary`
Screenshot 2023-09-01 at 3 32 13 PM

### - [ ] Plex isn't loading in the Search input
But it is loading in the dropdown, super weird! I highlighted the `as` I hope you can see the slight difference between Helvetica (Input) and Plex (dropdown)? 
Screenshot 2023-09-01 at 3 34 43 PM


### Alternate 
####  - [ ] Close button isn't a 48x48 square 
Not quite sure what causing this
Screenshot 2023-09-01 at 3 13 48 PM

#### - [ ] Dividers between options
Were you able to look into bringing in the dividers between the dropdown options for the alternate style? If not no big deal!

**Comment by annawen1** [comment]
@oliviaflory I believe things mentioned above should be fixed now!

**Review by oliviaflory — APPROVED** [review]
LGTM, thanks @annawen1 !

**Pull request merged**